src/tidecli/utils/file_handler.py

```diff
@@ -126,93 +127,54 @@ def create_task(task: TaskData, overwrite: bool, user_path: str | None = None) -
-            if f.task_directory:
-                f.saved_full_path = str(user_folder_root / f.task_directory / f.file_name)
-            else:
-                f.saved_full_path = str(user_folder / f.file_name)
-
-        save_files(task_files=task.supplementary_files, save_path=user_folder,
-                   overwrite=overwrite, msg="Supplementary files")
+        save_files(task_files=task.supplementary_files, folders=root_folders,
+                   msg="  supplementary files", overwrite=overwrite)
 
     if not saved:
         return False
 
-    write_metadata(
-        folder_path=user_folder_root,
-        metadata=task,
-    )
+    write_metadata(folder_path=user_folder_root, metadata=task)
 
     return saved
 
 
-def add_suffix(file_name: str, file_type: str) -> str:
-    """
-    Add suffix to file name if it is missing. Fix for C++ and C files.
-
-    TODO: Support more file types, should this be done in TIM?
-
-    :param file_name: Name of the file
-    :param file_type: Type of the file
-    :return: File name with suffix
+def save_files(task_files: list[TaskFile] | list[SupplementaryFile], folders: RootFolders,
+               msg: str, overwrite: bool = False) -> bool:
     """
-    if file_type == "c++" or file_type == "cpp":
-        return file_name + ".cpp"
-
-    if file_type == "cc":
-        return file_name + ".c"
-
-    return file_name
-
-
-def save_files(task_files: list[TaskFile] | list[SupplementaryFile], save_path: Path,
-               overwrite: bool = False, msg: str | None = None) -> bool:
-    """
-    Create files of tasks in the given path.
+    Save task files in the given path.
 
     :param task_files: Dict or list of dicts.
     Contain name with file extension (.eg .py or .txt ...) and content
-    :param save_path: Path to exercises in TIM
+    :param folders: Paths different scenarios (with or without task directory)
+    :param msg: Message to print when files are saved
     :param overwrite: Flag if overwrite
-    :param msg: Message to print if default is not ok
+    :return: True if files are saved, False if not
     """
-    # save_path.mkdir(parents=True, exist_ok=True)
 
     names = []
+    root_path = None  # None is never used, but it is needed to avoid linter error
 
     for f in task_files:
-        if f.saved_full_path:
-            file_path = Path(f.saved_full_path)
+        if f.task_directory:
+            root_path = folders.with_task_directory / f.task_directory
         else:
-            file_path = save_path / f.file_name
+            root_path = folders.without_task_directory
+
+        file_path = root_path / f.file_name
+        f.saved_absolute_file_name = str(file_path)
+
         if file_path.exists():
             if not overwrite:
                 click.echo(
                     f"File {file_path} already exists\n"
-                    f"To overwrite give tide task create -f {save_path}\n"
+                    f"To overwrite give add -f to previous command\n"
```

Comment:
"give" seems an extra verb here now.  Also, giving a full example is clearer as in the previous help text.

Reply:
typo from previous message, corrected